### ISSUE TITLE
[Bug Report] panic: runtime error: slice bounds out of range

### ISSUE THREAD

**Issue by Wikeolf** [daeuniverse/dae#357]
### Checks

- [X] I have searched the existing issues
- [X] I have read the documentation
- [X] Is it your first time sumbitting an issue

### Current Behavior

```
11月 15 11:13:23 wikeArch dae[5369]: level=info msg="localhost:38672 <-> 223.5.5.5:53" _qname=ogs.google.com. dialer=direct dscp=0 mac=">
11月 15 11:13:23 wikeArch dae[5369]: level=info msg="localhost:38672 <-> 223.5.5.5:53" _qname=ogs.google.com. dialer=direct dscp=0 mac=">
11月 15 11:13:23 wikeArch dae[5369]: panic: runtime error: slice bounds out of range [1347:1215]
11月 15 11:13:23 wikeArch dae[5369]: goroutine 3215 [running]:
11月 15 11:13:23 wikeArch dae[5369]: github.com/daeuniverse/dae/component/sniffing/internal/quicutils.ReassembleCryptoToBytesFromPool({0>
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/component/sniffing/internal/quicutils/relocation.go:88 +0x1d9
11月 15 11:13:23 wikeArch dae[5369]: github.com/daeuniverse/dae/component/sniffing.extractSniFromQuicPayload({0xc003360000?, 0x16?, 0x80>
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/component/sniffing/quic.go:180 +0x21c
11月 15 11:13:23 wikeArch dae[5369]: github.com/daeuniverse/dae/component/sniffing.sniffQuicBlock({0xc0033e6000, 0x55c201974bd8?, 0xc003>
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/component/sniffing/quic.go:149 +0x8f0
11月 15 11:13:23 wikeArch dae[5369]: github.com/daeuniverse/dae/component/sniffing.(*Sniffer).SniffQuic(0xc000bf5380?)
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/component/sniffing/quic.go:47 +0x45
11月 15 11:13:23 wikeArch dae[5369]: github.com/daeuniverse/dae/component/sniffing.sniffGroup({0xc00122f838, 0x1, 0x55c202693900?})
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/component/sniffing/sniffer.go:53 +0x47
11月 15 11:13:23 wikeArch dae[5369]: github.com/daeuniverse/dae/component/sniffing.(*Sniffer).SniffUdp(0xc00432be00)
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/component/sniffing/sniffer.go:112 +0x127
11月 15 11:13:23 wikeArch dae[5369]: github.com/daeuniverse/dae/control.(*ControlPlane).handlePkt(0xc00212f4a0, 0xc000480078, {0xc0033e6>
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/control/udp.go:145 +0x345
11月 15 11:13:23 wikeArch dae[5369]: github.com/daeuniverse/dae/control.(*ControlPlane).Serve.func5.1({0xc0033e6000, 0x51a, 0x800}, {0xc>
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/control/control_plane.go:778 +0x551
11月 15 11:13:23 wikeArch dae[5369]: created by github.com/daeuniverse/dae/control.(*ControlPlane).Serve.func5 in goroutine 218
11月 15 11:13:23 wikeArch dae[5369]:         github.com/daeuniverse/dae/control/control_plane.go:737 +0xec
11月 15 11:13:23 wikeArch systemd[1]: dae.service: Main process exited, code=exited, status=2/INVALIDARGUMENT

```

will lost any connection. reboot and not enable dae service is fine. but will crash when dae start.

### Expected Behavior

no more crash

### Steps to Reproduce

1. Archlinux
2. Always runs fine with normal configuration, but a rare crash occurs.

### Environment

- **Dae version (use `dae --version`)**: dae version 0.3.0 
- **OS (e.g `cat /etc/os-release`)**: Arch Linux
- **Kernel (e.g. `uname -a`)**: `Linux wikeArch 6.6.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 08 Nov 2023 16:05:38 +0000 x86_64 GNU/Linux`
- **Others**: 
```
pacman -Qi dae                                                                                                    ok  11:24:03 上午 
名字           : dae
版本           : 0.3.0-1
描述           : A Linux lightweight and high-performance transparent proxy solution based on eBPF.
架构           : x86_64
URL            : https://github.com/daeuniverse/dae
软件许可       : AGPL
组             : 无
提供           : 无
依赖于         : glibc  v2ray-geoip  v2ray-domain-list-community
可选依赖       : 无
依赖它         : 无
被可选依赖     : 无
与它冲突       : 无
取代           : 无
安装后大小     : 22.79 MiB
打包者         : lilac (on behalf of integral) <integral@murena.io>
编译日期       : 2023年09月03日 星期日 06时40分44秒
安装日期       : 2023年11月15日 星期三 11时12分16秒
安装原因       : 单独指定安装
安装脚本       : 否
验证者         : 数字签名
```

### Anything else?

if difficult to reproduce, I've kept the runtime environment here to work with the fixes

**Comment by kyleli666** [comment]
和我一样的问题，可以试一下这里面提到的版本。
https://github.com/daeuniverse/dae/issues/354

**Comment by Wikeolf** [comment]
> 和我一样的问题，可以试一下这里面提到的版本。 #354

Already to switch this version temporarily, but hopefully this will be fixed on a future release

**Comment by Wikeolf** [comment]
Well, I found that developers are trying fix this issue in #301.This issue will be closed, at a later date I will try to compile the latest version from source, hopefully dae gets better and better!

**Comment by Wikeolf** [comment]
> 和我一样的问题，可以试一下这里面提到的版本。 #354

The version built from the latest source code seems to have fixed this problem, you can try it.
